// @capgo/capacitor-document-scanner/dist/esm/definitions.js
export var ScannerMode;
(function (ScannerMode) {
    /**
     * Basic document scanning with crop and rotate features only.
     * No filters or ML-based enhancements.
     */
    ScannerMode["Base"] = "base";
    /**
     * Basic features plus automatic filters (grayscale, auto-enhancement).
     */
    ScannerMode["BaseWithFilter"] = "base_with_filter";
    /**
     * Full feature set including ML-based image cleaning.
     * Automatically removes stains, fingers, and other artifacts.
     */
    ScannerMode["Full"] = "full";
})(ScannerMode || (ScannerMode = {}));
export var ResponseType;
(function (ResponseType) {
    /**
     * Return scanned images as base64-encoded strings.
     */
    ResponseType["Base64"] = "base64";
    /**
     * Return scanned images as file paths on disk.
     */
    ResponseType["ImageFilePath"] = "imageFilePath";
})(ResponseType || (ResponseType = {}));
export var ScanDocumentResponseStatus;
(function (ScanDocumentResponseStatus) {
    /**
     * The scan completed successfully.
     */
    ScanDocumentResponseStatus["Success"] = "success";
    /**
     * The user cancelled the scan flow.
     */
    ScanDocumentResponseStatus["Cancel"] = "cancel";
})(ScanDocumentResponseStatus || (ScanDocumentResponseStatus = {}));
