// @capgo/capacitor-document-scanner/dist/plugin.js
var capacitorDocumentScanner = (function (exports, core) {
    'use strict';

    exports.ScannerMode = void 0;
    (function (ScannerMode) {
        /**
         * Basic document scanning with crop and rotate features only.
         * No filters or ML-based enhancements.
         */
        ScannerMode["Base"] = "base";
        /**
         * Basic features plus automatic filters (grayscale, auto-enhancement).
         */
        ScannerMode["BaseWithFilter"] = "base_with_filter";
        /**
         * Full feature set including ML-based image cleaning.
         * Automatically removes stains, fingers, and other artifacts.
         */
        ScannerMode["Full"] = "full";
    })(exports.ScannerMode || (exports.ScannerMode = {}));
    exports.ResponseType = void 0;
    (function (ResponseType) {
        /**
         * Return scanned images as base64-encoded strings.
         */
        ResponseType["Base64"] = "base64";
        /**
         * Return scanned images as file paths on disk.
         */
        ResponseType["ImageFilePath"] = "imageFilePath";
    })(exports.ResponseType || (exports.ResponseType = {}));
    exports.ScanDocumentResponseStatus = void 0;
    (function (ScanDocumentResponseStatus) {
        /**
         * The scan completed successfully.
         */
        ScanDocumentResponseStatus["Success"] = "success";
        /**
         * The user cancelled the scan flow.
         */
        ScanDocumentResponseStatus["Cancel"] = "cancel";
    })(exports.ScanDocumentResponseStatus || (exports.ScanDocumentResponseStatus = {}));

    const DocumentScanner = core.registerPlugin('DocumentScanner', {
        web: () => Promise.resolve().then(function () { return web; }).then((m) => new m.DocumentScannerWeb()),
    });

    class DocumentScannerWeb extends core.WebPlugin {
        async scanDocument(_options) {
            throw this.unimplemented('Document scanning is not supported on the web.');
        }
        async getPluginVersion() {
            return { version: 'web' };
        }
    }

    var web = /*#__PURE__*/Object.freeze({
        __proto__: null,
        DocumentScannerWeb: DocumentScannerWeb
    });

    exports.DocumentScanner = DocumentScanner;

    return exports;

})({}, capacitorExports);
